DSharpPlus/Utilities.cs

```diff
@@ -169,14 +169,116 @@ internal static IEnumerable<ulong> GetChannelMentions(DiscordMessage message)
         }
 
-        internal static IEnumerable<ulong> GetEmojis(DiscordMessage message)
+        internal static IEnumerable<ulong> GetGuildEmojis(DiscordMessage message)
         {
             var regex = new Regex(@"<a?:([a-zA-Z0-9_]+):(\d+)>", RegexOptions.ECMAScript);
             var matches = regex.Matches(message.Content);
             foreach (Match match in matches)
                 yield return ulong.Parse(match.Groups[2].Value, CultureInfo.InvariantCulture);
         }
 
+        public static IEnumerable<DiscordEmoji> GetCommonEmojis(DiscordMessage message)
+        {
+            return DiscordEmoji.UnicodeEmojis.Keys
+                .Where(e => message.Content.Contains(e))
+                .Select(e => DiscordEmoji.FromName(message.Discord, e, false))
+                .Union(DiscordEmoji.DiscordNameLookup.Keys
+                    .Where(e => message.Content.Contains(e))
+                    .Select(e => DiscordEmoji.FromUnicode(e)))
+                .ToList();
+        }
+
+        /// <summary>
+        /// Converts all of the Discord message snowflake tags to a human-readable format
+        /// </summary>
+        /// <param name="message">The message to convert</param>
+        /// <returns>A human-readable format of the message</returns>
+        public static Task<string> HumanizeContentAsync(DiscordMessage message)
+            => HumanizeContentAsync(message, (e)
+                => e.IsUnicode ? Task.FromResult(e.Name) : Task.FromResult(string.Empty));
+
+        /// <summary>
+        /// Converts all of the Discord message snowflake tags to a human-readable format
+        /// </summary>
+        /// <param name="message">The message to convert</param>
+        /// <param name="emojiReplacer">A callback to specify how emojis should be converted</param>
+        /// <returns>A human-readable format of the message</returns>
+        public async static Task<string> HumanizeContentAsync(DiscordMessage message, Func<DiscordEmoji, Task<string>> emojiReplacer)
+        {
+            // First we take care of "special tags (<>), including guild emojies"
+            var regex = new Regex(@"<(?<TagType>(@[!&]?)|(#)|(a?:([a-zA-Z0-9_]+):))(?<TagId>\d+)>", RegexOptions.ECMAScript);
+            var text = await regex.ReplaceAsync(message.Content, async m =>
+            {
+                var tagType = m.Groups["TagType"].Value;
+                var tagId = ulong.Parse(m.Groups["TagId"].Value);
+                switch (tagType)
+                {
+                    case "@": // Username mention
+                        return $"@{message.MentionedUsers.FirstOrDefault(u => u.Id == tagId).Username}";
+
+                    case "@!": // Nickname mention
+                        var user = message.MentionedUsers.FirstOrDefault(u => u.Id == tagId);
+                        if (user is DiscordMember member)
+                        {
+                            var nickname = string.IsNullOrEmpty(member.Nickname) ?
```

Comment:
You can use DisplayName

Reply:
It's guaranteed to be the nickname in case it is set?

Reply:
Yes